retrieval/scrape_test.go

```diff
@@ -627,6 +629,49 @@ func TestScrapeLoopAppend(t *testing.T) {
 	}
 }
 
+func TestScrapeLoopAppendSampleLimit(t *testing.T) {
+	resApp := &collectResultAppender{}
+	app := &limitAppender{Appender: resApp, limit: 1}
+
+	sl := newScrapeLoop(context.Background(),
+		nil, nil, nil,
+		nopMutator,
+		nopMutator,
+		func() storage.Appender { return app },
+	)
+
+	now := time.Now()
+	_, _, err := sl.append([]byte("metric_a 1\nmetric_b 1\nmetric_c 1\n"), now)
+	if err != errSampleLimit {
+		t.Fatalf("Did not see expected sample limit error: %s", err)
+	}
+
+	// Check that the Counter has been incremented a simgle time for the scrape,
+	// not multiple times for each sample
+	metric := dto.Metric{}
+	err = targetScrapeSampleLimit.Write(&metric)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	value := metric.GetCounter().GetValue()
+	if value != 1 {
```

Comment:
This tests that its value is 1, depending on test order that may not be the case. Test that is increased by 1.